`fatal error: oxenc/endian.h: No such file or directory`

**Describe the bug**
Trying to build on Fedora, I get the above error.

**To Reproduce**
```
git clone --recursive https://github.com/oxen-io/lokinet
cd lokinet
mkdir build
cd build
cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DWITH_LTO=OFF
make -j$(nproc)
```

**Screenshots Or Logs**

<details>
<summary>Log</summary>
<pre><code>
~> build (dev) cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DWITH_LTO=OFF
-- Enabling ccache for C
-- Enabling ccache for CXX
-- The C compiler identification is GNU 11.3.1
-- The CXX compiler identification is GNU 11.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- LTO disabled
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Have working 64bit atomics
-- Performing Test was_compiled
-- Performing Test was_compiled - Success
-- we have std::optional
-- Performing Test filesystem_compiled
-- Performing Test filesystem_compiled - Success
-- No extra link flag needed for std::filesystem
-- we have std::filesystem
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'libuv>=1.18.0'
--   Found libuv, version 1.44.1
-- Checking for module 'libsodium>=1.0.18'
--   Found libsodium, version 1.0.18
-- Checking for module 'liboxenc>=1.0.2'
--   Package 'liboxenc', required by 'virtual:world', not found
-- using oxen-encoding submodule
-- oxenc v1.0.2
-- Checking for module 'liboxenmq>=1.2.4'
--   Found liboxenmq, version 1.2.5
-- Found system liboxenmq 1.2.5
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking for module 'libunbound'
--   Found libunbound, version 1.13.2
-- Checking for module 'libsystemd'
--   Found libsystemd, version 249
-- Found Git: /usr/bin/git (found version "2.35.3") 
-- Checking submodules
-- Submodule 'external/nlohmann' is up-to-date
-- Submodule 'external/cxxopts' is up-to-date
-- Submodule 'external/ghc-filesystem' is up-to-date
-- Submodule 'external/date' is up-to-date
-- Submodule 'external/pybind11' is up-to-date
-- Submodule 'external/sqlite_orm' is up-to-date
-- Submodule 'external/oxen-mq' is up-to-date
-- Submodule 'external/oxen-encoding' is up-to-date
-- Submodule 'external/uvw' is up-to-date
-- Submodule 'external/cpr' is up-to-date
-- Submodule 'external/ngtcp2' is up-to-date
-- Using the single-header code from /home/paveloom/Software/lokinet-copy/external/nlohmann/single_include/
-- cxxopts version 2.2.0
# date: USE_SYSTEM_TZ_DB OFF
# date: MANUAL_TZ_DB OFF
# date: USE_TZ_DB_IN_DOT OFF
# date: BUILD_SHARED_LIBS OFF
# date: ENABLE_DATE_TESTING OFF
# date: DISABLE_STRING_VIEW OFF
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.36.0
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - not found
-- Looking for endian.h
-- Looking for endian.h - found
-- Looking for byteswap.h
-- Looking for byteswap.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Looking for be64toh
-- Looking for be64toh - not found
-- Looking for bswap_64
-- Looking for bswap_64 - found
-- Found CURL: /usr/lib64/libcurl.so (found version "7.79.1") found components: HTTP HTTPS SSL 
-- Checking for module 'jemalloc'
--   Found jemalloc, version 5.2.1_0
-- Performing Test COMPILER_SUPPORTS_AVX2
-- Performing Test COMPILER_SUPPORTS_AVX2 - Success
-- Performing Test COMPILER_SUPPORTS_FMA
-- Performing Test COMPILER_SUPPORTS_FMA - Success
-- Building libntrup with runtime AVX2/FMA support
-- Found Git: /usr/bin/git
-- Checking for module 'libcrypt'
--   Found libcrypt, version 4.4.28
-- using libcrypt 4.4.28
-- Found setcap binary: /sbin/setcap
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1n")  
-- Documentation generation disabled (mkdocs not found)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/paveloom/Software/lokinet-copy/build
~> build (dev) make -j(nproc)
[  0%] Building CXX object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ntru.cpp.o
[  0%] Generating constants/version.cpp
[  0%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pkt.c.o
[  0%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/auth.cpp.o
-- You are currently on commit bad98b54
-- You are not building a tagged release; setting version tag to 'bad98b54'
[  0%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomsmall.c.o
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conv.c.o
[  1%] Built target genversion
[  1%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/bearer.cpp.o
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/swap.c.o
[  3%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cookies.cpp.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_str.c.o
[  3%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_round3.c.o
[  3%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curl_container.cpp.o
[  3%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cprtypes.cpp.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_vec.c.o
[  3%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_recip3.c.o
[  4%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_buf.c.o
[  5%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/small.c.o
[  7%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curlholder.cpp.o
[  7%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_mult.c.o
[  7%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conn.c.o
[  7%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomweightw.c.o
[  7%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/error.cpp.o
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_conn.c: In function ‘conn_write_pkt’:
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘scid’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3371 |     ngtcp2_pkt_hd_init(hd, hd_flags, type, &conn->dcid.current.cid, scid,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3372 |                        pktns->tx.last_pkt_num + 1,
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3373 |                        pktns_select_pkt_numlen(pktns), conn->version, 0);
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘hd_flags’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/random32.c.o
[  8%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_mem.c.o
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/dec.c.o
[  8%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/multipart.cpp.o
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_mult.c.o
[  9%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_recip.c.o
[ 10%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/parameters.cpp.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pq.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/keypair.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_rounded.c.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_map.c.o
[ 11%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/payload.cpp.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/int32_sort.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/enc.c.o
[ 12%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rob.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq.c.o
[ 14%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomsmall.c.o
[ 14%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/proxies.cpp.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ppe.c.o
[ 15%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/weight.c.o
[ 15%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/swap.c.o
[ 16%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/response.cpp.o
[ 16%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_crypto.c.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_round3.c.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_recip3.c.o
/home/paveloom/Software/lokinet-copy/external/cpr/cpr/response.cpp: In member function ‘std::vector<std::__cxx11::basic_string<char> > cpr::Response::GetCertInfo()’:
/home/paveloom/Software/lokinet-copy/external/cpr/cpr/response.cpp:38:26: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   38 |     for (size_t i = 0; i < ci->num_of_certs; i++) {
      |                        ~~^~~~~~~~~~~~~~~~~~
[ 17%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_err.c.o
[ 17%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/session.cpp.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/small.c.o
[ 18%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomweightw.c.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_range.c.o
[ 20%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/timeout.cpp.o
[ 20%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/dec.c.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rtb.c.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_acktr.c.o
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_rtb.c: In function ‘ngtcp2_rtb_recv_ack’:
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_rtb.c:617:16: warning: ‘pktns’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  617 |     *pbufchain = bufchain->next;
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/paveloom/Software/lokinet-copy/external/ngtcp2/lib/ngtcp2_rtb.c:641:17: note: ‘pktns’ was declared here
  641 |   ngtcp2_pktns *pktns;
      |                 ^~~~~
[ 20%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/r3_recip.c.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_strm.c.o
[ 22%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/unix_socket.cpp.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_idtr.c.o
[ 23%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_gaptr.c.o
[ 23%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/keypair.c.o
[ 23%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/util.cpp.o
[ 23%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_rounded.c.o
[ 24%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ringbuf.c.o
[ 24%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/mult.c.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/enc.c.o
[ 25%] Linking CXX static library libcpr.a
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_log.c.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/int32_sort.c.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_qlog.c.o
[ 27%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_mod3.c.o
[ 28%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cid.c.o
[ 28%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ksl.c.o
[ 28%] Linking CXX static library liblokinet-cryptography.a
[ 28%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cc.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr2.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_addr.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_path.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pv.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_version.c.o
[ 30%] Built target cpr
[ 31%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rst.c.o
[ 31%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_window_filter.c.o
[ 31%] Built target lokinet-cryptography
[ 31%] Linking C static library libngtcp2_static.a
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/bencode.cpp.o
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/constants/version.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/fs.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/json.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/android_logger.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/buffer.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/file_logger.cpp.o
[ 34%] Built target ngtcp2_static
[ 35%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/logger.cpp.o
/home/paveloom/Software/lokinet-copy/llarp/util/buffer.cpp:2:10: fatal error: oxenc/endian.h: No such file or directory
    2 | #include <oxenc/endian.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [llarp/CMakeFiles/lokinet-util.dir/build.make:109: llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1308: llarp/CMakeFiles/lokinet-util.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
</code></pre>
</details>

**Device and Operating system (please complete the following information):**
 - OS: Fedora Linux
-  Device: PC
 - Lokinet Version number or Git commit hash: https://github.com/oxen-io/lokinet/commit/bad98b547670cb99c3835d7a672a4078f28216c5

<details>
<summary>I get this:</summary>
<pre><code>
~> build (issue-1911-fixup-oxenc-linkage) cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DWITH_LTO=OFF
-- Enabling ccache for C
-- Enabling ccache for CXX
-- The C compiler identification is GNU 11.3.1
-- The CXX compiler identification is GNU 11.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- LTO disabled
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Have working 64bit atomics
-- Performing Test was_compiled
-- Performing Test was_compiled - Success
-- we have std::optional
-- Performing Test filesystem_compiled
-- Performing Test filesystem_compiled - Success
-- No extra link flag needed for std::filesystem
-- we have std::filesystem
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'libuv>=1.18.0'
--   Found libuv, version 1.44.1
-- Checking for module 'libsodium>=1.0.18'
--   Found libsodium, version 1.0.18
-- Checking for module 'liboxenc>=1.0.2'
--   Package 'liboxenc', required by 'virtual:world', not found
-- using oxen-encoding submodule
-- oxenc v1.0.2
-- Checking for module 'liboxenmq>=1.2.4'
--   Found liboxenmq, version 1.2.5
-- Found system liboxenmq 1.2.5
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking for module 'libunbound'
--   Found libunbound, version 1.13.2
-- Checking for module 'libsystemd'
--   Found libsystemd, version 249
-- Found Git: /usr/bin/git (found version "2.35.3") 
-- Checking submodules
-- Submodule 'external/nlohmann' is up-to-date
-- Submodule 'external/cxxopts' is up-to-date
-- Submodule 'external/ghc-filesystem' is up-to-date
-- Submodule 'external/date' is up-to-date
-- Submodule 'external/pybind11' is up-to-date
-- Submodule 'external/sqlite_orm' is up-to-date
-- Submodule 'external/oxen-mq' is up-to-date
-- Submodule 'external/oxen-encoding' is up-to-date
-- Submodule 'external/uvw' is up-to-date
-- Submodule 'external/cpr' is up-to-date
-- Submodule 'external/ngtcp2' is up-to-date
-- Using the single-header code from /home/paveloom/Software/loki-network-copy/external/nlohmann/single_include/
-- cxxopts version 2.2.0
# date: USE_SYSTEM_TZ_DB OFF
# date: MANUAL_TZ_DB OFF
# date: USE_TZ_DB_IN_DOT OFF
# date: BUILD_SHARED_LIBS OFF
# date: ENABLE_DATE_TESTING OFF
# date: DISABLE_STRING_VIEW OFF
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.36.0
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - not found
-- Looking for endian.h
-- Looking for endian.h - found
-- Looking for byteswap.h
-- Looking for byteswap.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Looking for be64toh
-- Looking for be64toh - not found
-- Looking for bswap_64
-- Looking for bswap_64 - found
-- Found CURL: /usr/lib64/libcurl.so (found version "7.79.1") found components: HTTP HTTPS SSL 
-- Checking for module 'jemalloc'
--   Found jemalloc, version 5.2.1_0
-- Performing Test COMPILER_SUPPORTS_AVX2
-- Performing Test COMPILER_SUPPORTS_AVX2 - Success
-- Performing Test COMPILER_SUPPORTS_FMA
-- Performing Test COMPILER_SUPPORTS_FMA - Success
-- Building libntrup with runtime AVX2/FMA support
-- Found Git: /usr/bin/git
-- Checking for module 'libcrypt'
--   Found libcrypt, version 4.4.28
-- using libcrypt 4.4.28
-- Found setcap binary: /sbin/setcap
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1n")  
-- Documentation generation disabled (mkdocs not found)
-- Configuring done
CMake Error at llarp/CMakeLists.txt:3 (add_library):
  Target "lokinet-util" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at llarp/CMakeLists.txt:44 (add_library):
  Target "lokinet-platform" links to target "oxenc::oxenc" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at llarp/CMakeLists.txt:89 (add_library):
  Target "liblokinet" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at llarp/CMakeLists.txt:89 (add_library):
  Target "liblokinet" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at llarp/CMakeLists.txt:259 (add_library):
  Target "lokinet-shared" links to target "oxenc::oxenc" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at llarp/CMakeLists.txt:259 (add_library):
  Target "lokinet-shared" links to target "oxenc::oxenc" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:2 (add_executable):
  Target "lokinet-vpn" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:2 (add_executable):
  Target "lokinet-vpn" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:7 (add_executable):
  Target "lokinet" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:7 (add_executable):
  Target "lokinet" links to target "oxenc::oxenc" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:11 (add_executable):
  Target "lokinet-bootstrap" links to target "oxenc::oxenc" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at daemon/CMakeLists.txt:11 (add_executable):
  Target "lokinet-bootstrap" links to target "oxenc::oxenc" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
~> build (issue-1911-fixup-oxenc-linkage) make -j(nproc)
[  0%] Generating constants/version.cpp
[  0%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pkt.c.o
[  0%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/auth.cpp.o
[  0%] Building CXX object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ntru.cpp.o
-- You are currently on commit 9424a7f8
-- You are not building a tagged release; setting version tag to '9424a7f8'
[  0%] Built target genversion
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conv.c.o
[  1%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomsmall.c.o
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_str.c.o
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/swap.c.o
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_round3.c.o
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_recip3.c.o
[  2%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/bearer.cpp.o
[  2%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_vec.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_buf.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conn.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_mem.c.o
[  4%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/small.c.o
[  5%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pq.c.o
[  7%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cookies.cpp.o
[  7%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_map.c.o
[  7%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_mult.c.o
[  7%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rob.c.o
[  7%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomweightw.c.o
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/random32.c.o
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/dec.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ppe.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_crypto.c.o
[  9%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_mult.c.o
[ 10%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_recip.c.o
[ 10%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_err.c.o
[ 10%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cprtypes.cpp.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_range.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/keypair.c.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_acktr.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_rounded.c.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rtb.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/enc.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/int32_sort.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq.c.o
[ 14%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomsmall.c.o
[ 14%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/weight.c.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c: In function ‘ngtcp2_rtb_recv_ack’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:617:16: warning: ‘pktns’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  617 |     *pbufchain = bufchain->next;
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:641:17: note: ‘pktns’ was declared here
  641 |   ngtcp2_pktns *pktns;
      |                 ^~~~~
[ 14%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curl_container.cpp.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_strm.c.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_idtr.c.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_gaptr.c.o
[ 16%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ringbuf.c.o
[ 16%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/swap.c.o
[ 16%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_log.c.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_round3.c.o
[ 17%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_qlog.c.o
[ 18%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curlholder.cpp.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c: In function ‘conn_write_pkt’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘scid’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3371 |     ngtcp2_pkt_hd_init(hd, hd_flags, type, &conn->dcid.current.cid, scid,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3372 |                        pktns->tx.last_pkt_num + 1,
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3373 |                        pktns_select_pkt_numlen(pktns), conn->version, 0);
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘hd_flags’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cid.c.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ksl.c.o
[ 20%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_recip3.c.o
[ 20%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/error.cpp.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cc.c.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr.c.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr2.c.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_addr.c.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_path.c.o
[ 22%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/small.c.o
[ 22%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/multipart.cpp.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pv.c.o
[ 23%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/parameters.cpp.o
[ 23%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_version.c.o
[ 24%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rst.c.o
[ 24%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_window_filter.c.o
[ 24%] Linking C static library libngtcp2_static.a
[ 24%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/payload.cpp.o
[ 24%] Built target ngtcp2_static
[ 24%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/proxies.cpp.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomweightw.c.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/dec.c.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/r3_recip.c.o
[ 27%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/response.cpp.o
[ 27%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/session.cpp.o
[ 28%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/keypair.c.o
[ 28%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_rounded.c.o
[ 28%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/timeout.cpp.o
[ 29%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/unix_socket.cpp.o
[ 29%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/mult.c.o
[ 29%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/util.cpp.o
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp: In member function ‘std::vector<std::__cxx11::basic_string<char> > cpr::Response::GetCertInfo()’:
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp:38:26: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   38 |     for (size_t i = 0; i < ci->num_of_certs; i++) {
      |                        ~~^~~~~~~~~~~~~~~~~~
[ 30%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/enc.c.o
[ 30%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/int32_sort.c.o
[ 30%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq.c.o
[ 31%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_mod3.c.o
[ 31%] Linking CXX static library liblokinet-cryptography.a
[ 31%] Built target lokinet-cryptography
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/bencode.cpp.o
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/constants/version.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/fs.cpp.o
[ 32%] Linking CXX static library libcpr.a
[ 32%] Built target cpr
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/json.cpp.o
/home/paveloom/Software/loki-network-copy/llarp/util/buffer.cpp:2:10: fatal error: oxenc/endian.h: No such file or directory
    2 | #include <oxenc/endian.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [llarp/CMakeFiles/lokinet-util.dir/build.make:109: llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1308: llarp/CMakeFiles/lokinet-util.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
</code></pre>
</details>

On Tue, 10 May 2022 07:07:35 -0700
Pavel Sobolev ***@***.***> wrote:

> <details>
> <summary>I get this:</summary>
> <pre><code>
> ~> build (issue-1911-fixup-oxenc-linkage) cmake ..
> -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DWITH_LTO=OFF --
> Enabling ccache for C -- Enabling ccache for CXX
> -- The C compiler identification is GNU 11.3.1
> -- The CXX compiler identification is GNU 11.3.1
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working C compiler: /usr/bin/cc - skipped
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++ - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- LTO disabled
> -- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
> -- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
> -- Have working 64bit atomics
> -- Performing Test was_compiled
> -- Performing Test was_compiled - Success
> -- we have std::optional
> -- Performing Test filesystem_compiled
> -- Performing Test filesystem_compiled - Success
> -- No extra link flag needed for std::filesystem
> -- we have std::filesystem
> -- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
> -- Checking for module 'libuv>=1.18.0'
> --   Found libuv, version 1.44.1
> -- Checking for module 'libsodium>=1.0.18'
> --   Found libsodium, version 1.0.18
> -- Checking for module 'liboxenc>=1.0.2'
> --   Package 'liboxenc', required by 'virtual:world', not found
> -- using oxen-encoding submodule
> -- oxenc v1.0.2
> -- Checking for module 'liboxenmq>=1.2.4'
> --   Found liboxenmq, version 1.2.5
> -- Found system liboxenmq 1.2.5
> -- Looking for pthread.h
> -- Looking for pthread.h - found
> -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
> -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
> -- Found Threads: TRUE  
> -- Checking for module 'libunbound'
> --   Found libunbound, version 1.13.2
> -- Checking for module 'libsystemd'
> --   Found libsystemd, version 249
> -- Found Git: /usr/bin/git (found version "2.35.3") 
> -- Checking submodules
> -- Submodule 'external/nlohmann' is up-to-date
> -- Submodule 'external/cxxopts' is up-to-date
> -- Submodule 'external/ghc-filesystem' is up-to-date
> -- Submodule 'external/date' is up-to-date
> -- Submodule 'external/pybind11' is up-to-date
> -- Submodule 'external/sqlite_orm' is up-to-date
> -- Submodule 'external/oxen-mq' is up-to-date
> -- Submodule 'external/oxen-encoding' is up-to-date
> -- Submodule 'external/uvw' is up-to-date
> -- Submodule 'external/cpr' is up-to-date
> -- Submodule 'external/ngtcp2' is up-to-date
> -- Using the single-header code from
> /home/paveloom/Software/loki-network-copy/external/nlohmann/single_include/
> -- cxxopts version 2.2.0 # date: USE_SYSTEM_TZ_DB OFF
> # date: MANUAL_TZ_DB OFF
> # date: USE_TZ_DB_IN_DOT OFF
> # date: BUILD_SHARED_LIBS OFF
> # date: ENABLE_DATE_TESTING OFF
> # date: DISABLE_STRING_VIEW OFF
> -- Checking for module 'sqlite3'
> --   Found sqlite3, version 3.36.0
> -- Looking for arpa/inet.h
> -- Looking for arpa/inet.h - found
> -- Looking for netinet/in.h
> -- Looking for netinet/in.h - found
> -- Looking for stddef.h
> -- Looking for stddef.h - found
> -- Looking for stdint.h
> -- Looking for stdint.h - found
> -- Looking for stdlib.h
> -- Looking for stdlib.h - found
> -- Looking for string.h
> -- Looking for string.h - found
> -- Looking for unistd.h
> -- Looking for unistd.h - found
> -- Looking for sys/endian.h
> -- Looking for sys/endian.h - not found
> -- Looking for endian.h
> -- Looking for endian.h - found
> -- Looking for byteswap.h
> -- Looking for byteswap.h - found
> -- Looking for sys/types.h
> -- Looking for sys/types.h - found
> -- Check size of ssize_t
> -- Check size of ssize_t - done
> -- Looking for be64toh
> -- Looking for be64toh - not found
> -- Looking for bswap_64
> -- Looking for bswap_64 - found
> -- Found CURL: /usr/lib64/libcurl.so (found version "7.79.1") found
> components: HTTP HTTPS SSL -- Checking for module 'jemalloc'
> --   Found jemalloc, version 5.2.1_0
> -- Performing Test COMPILER_SUPPORTS_AVX2
> -- Performing Test COMPILER_SUPPORTS_AVX2 - Success
> -- Performing Test COMPILER_SUPPORTS_FMA
> -- Performing Test COMPILER_SUPPORTS_FMA - Success
> -- Building libntrup with runtime AVX2/FMA support
> -- Found Git: /usr/bin/git
> -- Checking for module 'libcrypt'
> --   Found libcrypt, version 4.4.28
> -- using libcrypt 4.4.28
> -- Found setcap binary: /sbin/setcap
> -- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1n")  
> -- Documentation generation disabled (mkdocs not found)
> -- Configuring done
> CMake Error at llarp/CMakeLists.txt:3 (add_library):
>   Target "lokinet-util" links to target "oxenc::oxenc" but the target
> was not found.  Perhaps a find_package() call is missing for an
> IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at llarp/CMakeLists.txt:44 (add_library):
>   Target "lokinet-platform" links to target "oxenc::oxenc" but the
> target was not found.  Perhaps a find_package() call is missing for
> an IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at llarp/CMakeLists.txt:89 (add_library):
>   Target "liblokinet" links to target "oxenc::oxenc" but the target
> was not found.  Perhaps a find_package() call is missing for an
> IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at llarp/CMakeLists.txt:89 (add_library):
>   Target "liblokinet" links to target "oxenc::oxenc" but the target
> was not found.  Perhaps a find_package() call is missing for an
> IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at llarp/CMakeLists.txt:259 (add_library):
>   Target "lokinet-shared" links to target "oxenc::oxenc" but the
> target was not found.  Perhaps a find_package() call is missing for
> an IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at llarp/CMakeLists.txt:259 (add_library):
>   Target "lokinet-shared" links to target "oxenc::oxenc" but the
> target was not found.  Perhaps a find_package() call is missing for
> an IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:2 (add_executable):
>   Target "lokinet-vpn" links to target "oxenc::oxenc" but the target
> was not found.  Perhaps a find_package() call is missing for an
> IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:2 (add_executable):
>   Target "lokinet-vpn" links to target "oxenc::oxenc" but the target
> was not found.  Perhaps a find_package() call is missing for an
> IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:7 (add_executable):
>   Target "lokinet" links to target "oxenc::oxenc" but the target was
> not found.  Perhaps a find_package() call is missing for an IMPORTED
> target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:7 (add_executable):
>   Target "lokinet" links to target "oxenc::oxenc" but the target was
> not found.  Perhaps a find_package() call is missing for an IMPORTED
> target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:11 (add_executable):
>   Target "lokinet-bootstrap" links to target "oxenc::oxenc" but the
> target was not found.  Perhaps a find_package() call is missing for
> an IMPORTED target, or an ALIAS target is missing?
> 
> 
> CMake Error at daemon/CMakeLists.txt:11 (add_executable):
>   Target "lokinet-bootstrap" links to target "oxenc::oxenc" but the
> target was not found.  Perhaps a find_package() call is missing for
> an IMPORTED target, or an ALIAS target is missing?
> 
> 
> -- Generating done
> CMake Generate step failed.  Build files cannot be regenerated
> correctly. ~> build (issue-1911-fixup-oxenc-linkage) make -j(nproc)
> [  0%] Generating constants/version.cpp
> [  0%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pkt.c.o [
> 0%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/auth.cpp.o [  0%] Building CXX
> object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ntru.cpp.o --
> You are currently on commit 9424a7f8 -- You are not building a tagged
> release; setting version tag to '9424a7f8' [  0%] Built target
> genversion [  1%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conv.c.o [
> 1%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomsmall.c.o
> [  1%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_str.c.o [
> 2%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/swap.c.o
> [  2%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_round3.c.o
> [  2%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_recip3.c.o
> [  2%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/bearer.cpp.o [  2%] Building C
> object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_vec.c.o [
> 3%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_buf.c.o [
> 3%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conn.c.o [
> 3%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_mem.c.o [
> 4%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/small.c.o
> [  5%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pq.c.o [  7%]
> Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cookies.cpp.o
> [  7%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_map.c.o [
> 7%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_mult.c.o
> [  7%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rob.c.o [
> 7%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomweightw.c.o
> [  8%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/random32.c.o
> [  8%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/dec.c.o [
>  9%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ppe.c.o [
> 9%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_crypto.c.o [
> 9%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_mult.c.o
> [ 10%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_recip.c.o
> [ 10%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_err.c.o [
> 10%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/cprtypes.cpp.o [ 11%] Building C
> object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_range.c.o [
> 11%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/keypair.c.o
> [ 11%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_acktr.c.o [
> 11%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_rounded.c.o
> [ 11%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rtb.c.o [
> 12%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/enc.c.o [
> 12%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/int32_sort.c.o
> [ 12%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq.c.o [
> 14%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomsmall.c.o
> [ 14%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/weight.c.o
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:
> In function ‘ngtcp2_rtb_recv_ack’:
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:617:16:
> warning: ‘pktns’ may be used uninitialized in this function
> [-Wmaybe-uninitialized] 617 |     *pbufchain = bufchain->next; |
> ~~~~~~~~~~~^~~~~~~~~~~~~~~~
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:641:17:
> note: ‘pktns’ was declared here 641 |   ngtcp2_pktns *pktns; |
>          ^~~~~ [ 14%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/curl_container.cpp.o [ 15%]
> Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_strm.c.o [
> 15%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_idtr.c.o [
> 15%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_gaptr.c.o [
> 16%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ringbuf.c.o [
> 16%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/swap.c.o
> [ 16%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_log.c.o [
> 17%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_round3.c.o
> [ 17%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_qlog.c.o [
> 18%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/curlholder.cpp.o
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:
> In function ‘conn_write_pkt’:
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5:
> warning: ‘scid’ may be used uninitialized in this function
> [-Wmaybe-uninitialized] 3371 |     ngtcp2_pkt_hd_init(hd, hd_flags,
> type, &conn->dcid.current.cid, scid, |
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 3372 |                        pktns->tx.last_pkt_num + 1, |
>              ~~~~~~~~~~~~~~~~~~~~~~~~~~~ 3373 |
>  pktns_select_pkt_numlen(pktns), conn->version, 0); |
>        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5:
> warning: ‘hd_flags’ may be used uninitialized in this function
> [-Wmaybe-uninitialized] [ 20%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cid.c.o [
> 20%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ksl.c.o [
> 20%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_recip3.c.o
> [ 20%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/error.cpp.o [ 20%] Building C
> object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cc.c.o
> [ 21%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr.c.o [
> 21%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr2.c.o [
> 21%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_addr.c.o [
> 22%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_path.c.o [
> 22%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/small.c.o
> [ 22%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/multipart.cpp.o [ 22%] Building C
> object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pv.c.o
> [ 23%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/parameters.cpp.o [ 23%] Building
> C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_version.c.o [
> 24%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rst.c.o [
> 24%] Building C object
> external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_window_filter.c.o
> [ 24%] Linking C static library libngtcp2_static.a [ 24%] Building
> CXX object external/CMakeFiles/cpr.dir/cpr/cpr/payload.cpp.o [ 24%]
> Built target ngtcp2_static [ 24%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/proxies.cpp.o [ 25%] Building C
> object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomweightw.c.o
> [ 25%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/dec.c.o [
> 25%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/r3_recip.c.o
> [ 27%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/response.cpp.o [ 27%] Building
> CXX object external/CMakeFiles/cpr.dir/cpr/cpr/session.cpp.o [ 28%]
> Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/keypair.c.o
> [ 28%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_rounded.c.o
> [ 28%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/timeout.cpp.o [ 29%] Building CXX
> object external/CMakeFiles/cpr.dir/cpr/cpr/unix_socket.cpp.o [ 29%]
> Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/mult.c.o
> [ 29%] Building CXX object
> external/CMakeFiles/cpr.dir/cpr/cpr/util.cpp.o
> /home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp:
> In member function ‘std::vector<std::__cxx11::basic_string<char> >
> cpr::Response::GetCertInfo()’:
> /home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp:38:26:
> warning: comparison of integer expressions of different signedness:
> ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare] 38 |
>  for (size_t i = 0; i < ci->num_of_certs; i++) { |
>     ~~^~~~~~~~~~~~~~~~~~ [ 30%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/enc.c.o [
> 30%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/int32_sort.c.o
> [ 30%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq.c.o [
> 31%] Building C object
> crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_mod3.c.o
> [ 31%] Linking CXX static library liblokinet-cryptography.a [ 31%]
> Built target lokinet-cryptography [ 31%] Building CXX object
> llarp/CMakeFiles/lokinet-util.dir/util/bencode.cpp.o [ 31%] Building
> CXX object llarp/CMakeFiles/lokinet-util.dir/constants/version.cpp.o
> [ 32%] Building CXX object
> llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o [ 32%] Building
> CXX object llarp/CMakeFiles/lokinet-util.dir/util/fs.cpp.o [ 32%]
> Linking CXX static library libcpr.a [ 32%] Built target cpr [ 32%]
> Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/json.cpp.o
> /home/paveloom/Software/loki-network-copy/llarp/util/buffer.cpp:2:10:
> fatal error: oxenc/endian.h: No such file or directory 2 | #include
> <oxenc/endian.h> |          ^~~~~~~~~~~~~~~~ compilation terminated.
> make[2]: *** [llarp/CMakeFiles/lokinet-util.dir/build.make:109:
> llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o] Error 1 make[2]:
> *** Waiting for unfinished jobs.... make[1]: ***
> [CMakeFiles/Makefile2:1308: llarp/CMakeFiles/lokinet-util.dir/all]
> Error 2 make: *** [Makefile:136: all] Error 2 </code></pre> </details>
> 

force pushed, forgot to commit a part. 
it is in: bbc1b965eb2180e79d7384d4ba73a6c4a2afe448

Yep, that fixed it.

I'm still getting `error while loading shared libraries: liboxenmq.so.0: cannot open shared object file` though... Would you mind helping with that?

<details>
<summary>Log</summary>
<pre><code>
build (issue-1911-fixup-oxenc-linkage) cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DWITH_LTO=OFF
-- Enabling ccache for C
-- Enabling ccache for CXX
-- The C compiler identification is GNU 11.3.1
-- The CXX compiler identification is GNU 11.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- LTO disabled
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Have working 64bit atomics
-- Performing Test was_compiled
-- Performing Test was_compiled - Success
-- we have std::optional
-- Performing Test filesystem_compiled
-- Performing Test filesystem_compiled - Success
-- No extra link flag needed for std::filesystem
-- we have std::filesystem
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'libuv>=1.18.0'
--   Found libuv, version 1.44.1
-- Checking for module 'libsodium>=1.0.18'
--   Found libsodium, version 1.0.18
-- Checking for module 'liboxenc>=1.0.2'
--   Package 'liboxenc', required by 'virtual:world', not found
-- using oxen-encoding submodule
-- oxenc v1.0.2
-- Checking for module 'liboxenmq>=1.2.4'
--   Found liboxenmq, version 1.2.5
-- Found system liboxenmq 1.2.5
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking for module 'libunbound'
--   Found libunbound, version 1.13.2
-- Checking for module 'libsystemd'
--   Found libsystemd, version 249
-- Found Git: /usr/bin/git (found version "2.35.3") 
-- Checking submodules
-- Submodule 'external/nlohmann' is up-to-date
-- Submodule 'external/cxxopts' is up-to-date
-- Submodule 'external/ghc-filesystem' is up-to-date
-- Submodule 'external/date' is up-to-date
-- Submodule 'external/pybind11' is up-to-date
-- Submodule 'external/sqlite_orm' is up-to-date
-- Submodule 'external/oxen-mq' is up-to-date
-- Submodule 'external/oxen-encoding' is up-to-date
-- Submodule 'external/uvw' is up-to-date
-- Submodule 'external/cpr' is up-to-date
-- Submodule 'external/ngtcp2' is up-to-date
-- Using the single-header code from /home/paveloom/Software/loki-network-copy/external/nlohmann/single_include/
-- cxxopts version 2.2.0
# date: USE_SYSTEM_TZ_DB OFF
# date: MANUAL_TZ_DB OFF
# date: USE_TZ_DB_IN_DOT OFF
# date: BUILD_SHARED_LIBS OFF
# date: ENABLE_DATE_TESTING OFF
# date: DISABLE_STRING_VIEW OFF
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.36.0
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - not found
-- Looking for endian.h
-- Looking for endian.h - found
-- Looking for byteswap.h
-- Looking for byteswap.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Looking for be64toh
-- Looking for be64toh - not found
-- Looking for bswap_64
-- Looking for bswap_64 - found
-- Found CURL: /usr/lib64/libcurl.so (found version "7.79.1") found components: HTTP HTTPS SSL 
-- Checking for module 'jemalloc'
--   Found jemalloc, version 5.2.1_0
-- Performing Test COMPILER_SUPPORTS_AVX2
-- Performing Test COMPILER_SUPPORTS_AVX2 - Success
-- Performing Test COMPILER_SUPPORTS_FMA
-- Performing Test COMPILER_SUPPORTS_FMA - Success
-- Building libntrup with runtime AVX2/FMA support
-- Found Git: /usr/bin/git
-- Checking for module 'libcrypt'
--   Found libcrypt, version 4.4.28
-- using libcrypt 4.4.28
-- Found setcap binary: /sbin/setcap
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1n")  
-- Documentation generation disabled (mkdocs not found)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/paveloom/Software/loki-network-copy/build
~> build (issue-1911-fixup-oxenc-linkage) make -j(nproc)
[  0%] Generating constants/version.cpp
[  0%] Building CXX object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ntru.cpp.o
[  0%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pkt.c.o
[  0%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/auth.cpp.o
-- You are currently on commit bbc1b965
[  0%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomsmall.c.o
-- You are not building a tagged release; setting version tag to 'bbc1b965'
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conv.c.o
[  1%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/bearer.cpp.o
[  1%] Built target genversion
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/swap.c.o
[  3%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cookies.cpp.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_vec.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_str.c.o
[  3%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_round3.c.o
[  4%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cprtypes.cpp.o
[  4%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_buf.c.o
[  4%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conn.c.o
[  4%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_recip3.c.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c: In function ‘conn_write_pkt’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘scid’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3371 |     ngtcp2_pkt_hd_init(hd, hd_flags, type, &conn->dcid.current.cid, scid,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3372 |                        pktns->tx.last_pkt_num + 1,
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3373 |                        pktns_select_pkt_numlen(pktns), conn->version, 0);
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘hd_flags’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[  5%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/small.c.o
[  5%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_mult.c.o
[  5%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curl_container.cpp.o
[  5%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_mem.c.o
[  5%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomweightw.c.o
[  7%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/random32.c.o
[  8%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curlholder.cpp.o
[  8%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/dec.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pq.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_map.c.o
[  9%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/error.cpp.o
[  9%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_mult.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rob.c.o
[ 10%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ppe.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_recip.c.o
[ 11%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/multipart.cpp.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/keypair.c.o
[ 11%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_crypto.c.o
[ 11%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_rounded.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/enc.c.o
[ 14%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/parameters.cpp.o
[ 14%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_err.c.o
[ 14%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/int32_sort.c.o
[ 14%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq.c.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_range.c.o
[ 15%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/payload.cpp.o
[ 16%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomsmall.c.o
[ 16%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/weight.c.o
[ 16%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_acktr.c.o
[ 16%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/swap.c.o
[ 16%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/proxies.cpp.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_round3.c.o
[ 17%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rtb.c.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_recip3.c.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c: In function ‘ngtcp2_rtb_recv_ack’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:617:16: warning: ‘pktns’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  617 |     *pbufchain = bufchain->next;
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:641:17: note: ‘pktns’ was declared here
  641 |   ngtcp2_pktns *pktns;
      |                 ^~~~~
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/small.c.o
[ 18%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/response.cpp.o
[ 20%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_strm.c.o
[ 21%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomweightw.c.o
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp: In member function ‘std::vector<std::__cxx11::basic_string<char> > cpr::Response::GetCertInfo()’:
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp:38:26: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   38 |     for (size_t i = 0; i < ci->num_of_certs; i++) {
      |                        ~~^~~~~~~~~~~~~~~~~~
[ 21%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/dec.c.o
[ 21%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/session.cpp.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_idtr.c.o
[ 21%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/r3_recip.c.o
[ 21%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_gaptr.c.o
[ 21%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/timeout.cpp.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ringbuf.c.o
[ 23%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/keypair.c.o
[ 23%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_log.c.o
[ 24%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/unix_socket.cpp.o
[ 24%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/util.cpp.o
[ 24%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_rounded.c.o
[ 24%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_qlog.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cid.c.o
[ 25%] Linking CXX static library libcpr.a
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/mult.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ksl.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cc.c.o
[ 27%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/enc.c.o
[ 27%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/int32_sort.c.o
[ 27%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq.c.o
[ 28%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr.c.o
[ 29%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_mod3.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_addr.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr2.c.o
[ 29%] Linking CXX static library liblokinet-cryptography.a
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_path.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pv.c.o
[ 30%] Built target cpr
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_version.c.o
[ 31%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_window_filter.c.o
[ 31%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rst.c.o
[ 31%] Linking C static library libngtcp2_static.a
[ 31%] Built target lokinet-cryptography
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/constants/version.cpp.o
[ 31%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/bencode.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o
[ 32%] Built target ngtcp2_static
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/fs.cpp.o
[ 32%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/json.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/android_logger.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/buffer.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/file_logger.cpp.o
[ 35%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/logger.cpp.o
[ 35%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/logger_internal.cpp.o
[ 35%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/loglevel.cpp.o
[ 36%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/ostream_logger.cpp.o
[ 36%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/syslog_logger.cpp.o
[ 36%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/win32_logger.cpp.o
[ 37%] Building C object llarp/CMakeFiles/lokinet-util.dir/util/lokinet_init.c.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/mem.cpp.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/printer.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/str.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/thread/queue_manager.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/thread/threading.cpp.o
[ 40%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/time.cpp.o
[ 40%] Linking CXX static library liblokinet-util.a
[ 41%] Built target lokinet-util
[ 41%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_address.cpp.o
[ 41%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip.cpp.o
[ 42%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/ev/ev_libuv.cpp.o
[ 42%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/ev/ev.cpp.o
[ 43%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_packet.cpp.o
[ 43%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_range.cpp.o
[ 43%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/net.cpp.o
[ 44%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/net_int.cpp.o
[ 44%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/sock_addr.cpp.o
[ 44%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/packet_router.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/egres_packet_router.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/platform.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/linux/netns.cpp.o
[ 47%] Linking CXX static library liblokinet-platform.a
[ 47%] Built target lokinet-platform
[ 48%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/definition.cpp.o
[ 48%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/config.cpp.o
[ 49%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/key_manager.cpp.o
[ 49%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/ini.cpp.o
[ 49%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/message.cpp.o
[ 49%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/name.cpp.o
[ 50%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/question.cpp.o
[ 50%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/rr.cpp.o
[ 50%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/serialize.cpp.o
[ 51%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/server.cpp.o
[ 51%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/srv_data.cpp.o
[ 51%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/unbound_resolver.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/consensus/table.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/consensus/reachability_testing.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/bootstrap.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/context.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/crypto_libsodium.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/crypto.cpp.o
[ 55%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/encrypted_frame.cpp.o
[ 55%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/types.cpp.o
[ 55%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/context.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/dht.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/explorenetworkjob.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localtaglookup.cpp.o
[ 57%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localrouterlookup.cpp.o
[ 57%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localserviceaddresslookup.cpp.o
[ 57%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/message.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findintro.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findrouter.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotintro.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotrouter.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/pubintro.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findname.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotname.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/publishservicejob.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/recursiverouterlookup.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/serviceaddresslookup.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/taglookup.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/endpoint_base.cpp.o
[ 63%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/context.cpp.o
[ 63%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/endpoint.cpp.o
[ 63%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/exit_messages.cpp.o
[ 64%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/policy.cpp.o
[ 64%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/session.cpp.o
[ 64%] Building CXX object llarp/CMakeFiles/liblokinet.dir/handlers/exit.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/handlers/tun.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/hook/shell.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/iwp.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/linklayer.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/message_buffer.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/session.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/link_manager.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/session.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/server.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/dht_immediate.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/link_intro.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/link_message_parser.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay_commit.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay_status.cpp.o
[ 71%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/address_info.cpp.o
[ 71%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/exit_info.cpp.o
[ 71%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/traffic_policy.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/nodedb.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/ihophandler.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/path_context.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/path.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/pathbuilder.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/pathset.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/transit_hop.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/peerstats/peer_db.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/peerstats/types.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/pow.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/profiling.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/address.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/client.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/connection.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/endpoint.cpp.o
[ 78%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/null_crypto.cpp.o
[ 78%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/server.cpp.o
[ 78%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/stream.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/tunnel.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_contact.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_id.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_version.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/name.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/outbound_message_handler.cpp.o
[ 82%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/outbound_session_maker.cpp.o
[ 82%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/rc_lookup_handler.cpp.o
[ 82%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/rc_gossiper.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/router.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/route_poker.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/systemd_resolved.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/dht_message.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/message_parser.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_confirm_message.cpp.o
[ 85%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_latency_message.cpp.o
[ 85%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_transfer_message.cpp.o
[ 85%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/transfer_traffic_message.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/lokid_rpc_client.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/rpc_server.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/endpoint_rpc.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/address.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/async_key_exchange.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/auth.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/convotag.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/context.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint_state.cpp.o
[ 90%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint_util.cpp.o
[ 90%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint.cpp.o
[ 90%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/hidden_service_address_lookup.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/identity.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/info.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/intro_set.cpp.o
[ 92%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/intro.cpp.o
[ 92%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/lns_tracker.cpp.o
[ 92%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/lookup.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/outbound_context.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/protocol.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/protocol_type.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/router_lookup_job.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/sendcontext.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/session.cpp.o
[ 96%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/tag.cpp.o
[ 96%] Linking CXX static library liblokinet.a
[ 96%] Built target liblokinet
[ 97%] Building CXX object daemon/CMakeFiles/lokinet.dir/lokinet.cpp.o
[ 97%] Building CXX object daemon/CMakeFiles/lokinet-bootstrap.dir/lokinet-bootstrap.cpp.o
[ 97%] Building CXX object llarp/CMakeFiles/lokinet-shared.dir/lokinet_shared.cpp.o
[ 97%] Building CXX object daemon/CMakeFiles/lokinet-vpn.dir/lokinet-vpn.cpp.o
[ 98%] Linking CXX executable lokinet-bootstrap
[ 98%] Built target lokinet-bootstrap
[ 98%] Linking CXX executable lokinet
[ 98%] Built target lokinet
[ 98%] Linking CXX shared library liblokinet.so
[ 98%] Built target lokinet-shared
[100%] Linking CXX executable lokinet-vpn
[100%] Built target lokinet-vpn
~> build (issue-1911-fixup-oxenc-linkage) sudo make install
[sudo] password for paveloom: 
Consolidate compiler generated dependencies of target lokinet-cryptography
[ 12%] Built target lokinet-cryptography
[ 12%] Built target genversion
Consolidate compiler generated dependencies of target lokinet-util
[ 22%] Built target lokinet-util
Consolidate compiler generated dependencies of target lokinet-platform
[ 28%] Built target lokinet-platform
Consolidate compiler generated dependencies of target ngtcp2_static
[ 41%] Built target ngtcp2_static
Consolidate compiler generated dependencies of target liblokinet
[ 90%] Built target liblokinet
Consolidate compiler generated dependencies of target lokinet-shared
[ 90%] Built target lokinet-shared
Consolidate compiler generated dependencies of target lokinet-vpn
[ 91%] Built target lokinet-vpn
Consolidate compiler generated dependencies of target lokinet
[ 92%] Built target lokinet
Consolidate compiler generated dependencies of target cpr
[ 98%] Built target cpr
Consolidate compiler generated dependencies of target lokinet-bootstrap
[100%] Built target lokinet-bootstrap
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib64/liblokinet.so
-- Set runtime path of "/usr/local/lib64/liblokinet.so" to ""
-- Installing: /usr/local/bin/lokinet
-- Set runtime path of "/usr/local/bin/lokinet" to ""
-- Installing: /usr/local/bin/lokinet-vpn
-- Set runtime path of "/usr/local/bin/lokinet-vpn" to ""
-- Installing: /usr/local/bin/lokinet-bootstrap
-- Set runtime path of "/usr/local/bin/lokinet-bootstrap" to ""
~> build (issue-1911-fixup-oxenc-linkage) lokinet --help
lokinet: error while loading shared libraries: liboxenmq.so.0: cannot open shared object file: No such file or directory
</code></pre>
</details>

Now I'm getting `/usr/bin/ld: cannot find -lzmq` after reinstalling `zeromq-devel` and `cppzmq-devel`...

UPD: Huh, an invalid symlink...

```
~> build (issue-1911-fixup-oxenc-linkage) ls -l /usr/lib64/libzmq.so
lrwxrwxrwx 15 root 24 Jul  2021 /usr/lib64/libzmq.so -> libzmq.so.5.2.4
~> build (issue-1911-fixup-oxenc-linkage) ls -l /usr/lib64/libzmq.so.5.2.4
"/usr/lib64/libzmq.so.5.2.4": No such file or directory (os error 2)
```

Fixed by reinstalling the `zeromq` package.

So, `pkg-config` is saying that the `liboxenc` library is missing:

```
~> build (issue-1911-fixup-oxenc-linkage) PKG_CONFIG_PATH=/usr/local/lib64/pkgconfig pkg-config --libs liboxenmq
Package liboxenc was not found in the pkg-config search path.
Perhaps you should add the directory containing `liboxenc.pc'
to the PKG_CONFIG_PATH environment variable
Package 'liboxenc', required by 'liboxenmq', not found
~> build (issue-1911-fixup-oxenc-linkage) sudo find /usr/ -name "*liboxenc*"
~> build (issue-1911-fixup-oxenc-linkage)
```

<details>
<summary>Log</summary>
<pre><code>
~> build (issue-1911-fixup-oxenc-linkage) cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DWITH_LTO=OFF
-- Enabling ccache for C
-- Enabling ccache for CXX
-- The C compiler identification is GNU 11.3.1
-- The CXX compiler identification is GNU 11.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- LTO disabled
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Have working 64bit atomics
-- Performing Test was_compiled
-- Performing Test was_compiled - Success
-- we have std::optional
-- Performing Test filesystem_compiled
-- Performing Test filesystem_compiled - Success
-- No extra link flag needed for std::filesystem
-- we have std::filesystem
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.0") 
-- Checking for module 'libuv>=1.18.0'
--   Found libuv, version 1.44.1
-- Checking for module 'libsodium>=1.0.18'
--   Found libsodium, version 1.0.18
-- Checking for module 'liboxenc>=1.0.2'
--   Package 'liboxenc', required by 'virtual:world', not found
-- using oxen-encoding submodule
-- oxenc v1.0.2
-- Checking for module 'liboxenmq>=1.2.4'
--   Package 'liboxenmq', required by 'virtual:world', not found
-- using oxenmq submodule
-- oxenmq v1.2.11
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Checking for module 'libzmq>=4.3'
--   Found libzmq, version 4.3.4
-- Checking for module 'libunbound'
--   Found libunbound, version 1.13.2
-- Checking for module 'libsystemd'
--   Found libsystemd, version 249
-- Found Git: /usr/bin/git (found version "2.35.3") 
-- Checking submodules
-- Submodule 'external/nlohmann' is up-to-date
-- Submodule 'external/cxxopts' is up-to-date
-- Submodule 'external/ghc-filesystem' is up-to-date
-- Submodule 'external/date' is up-to-date
-- Submodule 'external/pybind11' is up-to-date
-- Submodule 'external/sqlite_orm' is up-to-date
-- Submodule 'external/oxen-mq' is up-to-date
-- Submodule 'external/oxen-encoding' is up-to-date
-- Submodule 'external/uvw' is up-to-date
-- Submodule 'external/cpr' is up-to-date
-- Submodule 'external/ngtcp2' is up-to-date
-- Using the single-header code from /home/paveloom/Software/loki-network-copy/external/nlohmann/single_include/
-- cxxopts version 2.2.0
# date: USE_SYSTEM_TZ_DB OFF
# date: MANUAL_TZ_DB OFF
# date: USE_TZ_DB_IN_DOT OFF
# date: BUILD_SHARED_LIBS ON
# date: ENABLE_DATE_TESTING OFF
# date: DISABLE_STRING_VIEW OFF
-- Checking for module 'sqlite3'
--   Found sqlite3, version 3.36.0
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stdlib.h
-- Looking for stdlib.h - found
-- Looking for string.h
-- Looking for string.h - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - not found
-- Looking for endian.h
-- Looking for endian.h - found
-- Looking for byteswap.h
-- Looking for byteswap.h - found
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Check size of ssize_t
-- Check size of ssize_t - done
-- Looking for be64toh
-- Looking for be64toh - not found
-- Looking for bswap_64
-- Looking for bswap_64 - found
-- Found CURL: /usr/lib64/libcurl.so (found version "7.79.1") found components: HTTP HTTPS SSL 
-- Checking for module 'jemalloc'
--   Found jemalloc, version 5.2.1_0
-- Performing Test COMPILER_SUPPORTS_AVX2
-- Performing Test COMPILER_SUPPORTS_AVX2 - Success
-- Performing Test COMPILER_SUPPORTS_FMA
-- Performing Test COMPILER_SUPPORTS_FMA - Success
-- Building libntrup with runtime AVX2/FMA support
-- Found Git: /usr/bin/git
-- Checking for module 'libcrypt'
--   Found libcrypt, version 4.4.28
-- using libcrypt 4.4.28
-- Found setcap binary: /sbin/setcap
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1n")  
-- Documentation generation disabled (mkdocs not found)
-- Configuring done
-- Generating done
-- Build files have been written to: /home/paveloom/Software/loki-network-copy/build
~> build (issue-1911-fixup-oxenc-linkage) make -j(nproc)
[  0%] Generating constants/version.cpp
[  0%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/address.cpp.o
[  1%] Building CXX object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ntru.cpp.o
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pkt.c.o
[  1%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomsmall.c.o
-- You are currently on commit bbc1b965
[  1%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conv.c.o
[  2%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/auth.cpp.o
-- You are not building a tagged release; setting version tag to 'bbc1b965'
[  2%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/swap.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_str.c.o
[  3%] Built target genversion
[  3%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/connections.cpp.o
[  3%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_round3.c.o
[  3%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_vec.c.o
[  4%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_recip3.c.o
[  4%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/auth.cpp.o
[  4%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/jobs.cpp.o
[  4%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_buf.c.o
[  4%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/small.c.o
[  4%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/bearer.cpp.o
[  5%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_conn.c.o
[  6%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/oxenmq.cpp.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c: In function ‘conn_write_pkt’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘scid’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 3371 |     ngtcp2_pkt_hd_init(hd, hd_flags, type, &conn->dcid.current.cid, scid,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3372 |                        pktns->tx.last_pkt_num + 1,
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 3373 |                        pktns_select_pkt_numlen(pktns), conn->version, 0);
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_conn.c:3371:5: warning: ‘hd_flags’ may be used uninitialized in this function [-Wmaybe-uninitialized]
[  6%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cookies.cpp.o
[  6%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_mult.c.o
[  6%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_mem.c.o
[  6%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/proxy.cpp.o
[  8%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/cprtypes.cpp.o
[  9%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/randomweightw.c.o
[  9%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pq.c.o
[  9%] Building CXX object external/oxen-mq/CMakeFiles/oxenmq.dir/oxenmq/worker.cpp.o
[  9%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/random32.c.o
[  9%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curl_container.cpp.o
[ 10%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_map.c.o
[ 10%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/dec.c.o
[ 11%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/curlholder.cpp.o
[ 11%] Linking CXX shared library liboxenmq.so
[ 12%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rob.c.o
[ 12%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_mult.c.o
[ 13%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/error.cpp.o
[ 13%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/r3_recip.c.o
[ 13%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ppe.c.o
[ 13%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/multipart.cpp.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_crypto.c.o
[ 15%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/keypair.c.o
[ 15%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/parameters.cpp.o
[ 15%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_err.c.o
[ 16%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq_rounded.c.o
[ 17%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/payload.cpp.o
[ 17%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_range.c.o
[ 17%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/enc.c.o
[ 18%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_acktr.c.o
[ 18%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/proxies.cpp.o
[ 18%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/int32_sort.c.o
[ 18%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rtb.c.o
[ 18%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/response.cpp.o
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c: In function ‘ngtcp2_rtb_recv_ack’:
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:617:16: warning: ‘pktns’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  617 |     *pbufchain = bufchain->next;
      |     ~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/paveloom/Software/loki-network-copy/external/ngtcp2/lib/ngtcp2_rtb.c:641:17: note: ‘pktns’ was declared here
  641 |   ngtcp2_pktns *pktns;
      |                 ^~~~~
[ 19%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/ref/rq.c.o
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp: In member function ‘std::vector<std::__cxx11::basic_string<char> > cpr::Response::GetCertInfo()’:
/home/paveloom/Software/loki-network-copy/external/cpr/cpr/response.cpp:38:26: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
   38 |     for (size_t i = 0; i < ci->num_of_certs; i++) {
      |                        ~~^~~~~~~~~~~~~~~~~~
[ 19%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_strm.c.o
[ 20%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/session.cpp.o
[ 20%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomsmall.c.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_idtr.c.o
[ 22%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/weight.c.o
[ 22%] Built target oxenmq
[ 22%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/timeout.cpp.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_gaptr.c.o
[ 22%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ringbuf.c.o
[ 23%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/swap.c.o
[ 24%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_log.c.o
[ 24%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/unix_socket.cpp.o
[ 25%] Building CXX object external/CMakeFiles/cpr.dir/cpr/cpr/util.cpp.o
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_round3.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_qlog.c.o
[ 25%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cid.c.o
[ 25%] Linking CXX static library libcpr.a
[ 25%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_recip3.c.o
[ 26%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_ksl.c.o
[ 26%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_cc.c.o
[ 27%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/small.c.o
[ 27%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_bbr2.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_addr.c.o
[ 29%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/randomweightw.c.o
[ 29%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_path.c.o
[ 29%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/dec.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_pv.c.o
[ 30%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_version.c.o
[ 31%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/r3_recip.c.o
[ 31%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_rst.c.o
[ 32%] Building C object external/ngtcp2/lib/CMakeFiles/ngtcp2_static.dir/ngtcp2_window_filter.c.o
[ 32%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/keypair.c.o
[ 32%] Built target cpr
[ 32%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_rounded.c.o
[ 32%] Linking C static library libngtcp2_static.a
[ 33%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/mult.c.o
[ 33%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/enc.c.o
[ 33%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/int32_sort.c.o
[ 33%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq_mod3.c.o
[ 34%] Building C object crypto/CMakeFiles/lokinet-cryptography.dir/libntrup/src/avx/rq.c.o
[ 34%] Linking CXX static library liblokinet-cryptography.a
[ 34%] Built target ngtcp2_static
[ 34%] Built target lokinet-cryptography
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/constants/version.cpp.o
[ 34%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/bencode.cpp.o
[ 36%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/buffer.cpp.o
[ 36%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/fs.cpp.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/json.cpp.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/android_logger.cpp.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/buffer.cpp.o
[ 37%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/file_logger.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/logger.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/logger_internal.cpp.o
[ 38%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/loglevel.cpp.o
[ 39%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/ostream_logger.cpp.o
[ 39%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/syslog_logger.cpp.o
[ 39%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/logging/win32_logger.cpp.o
[ 40%] Building C object llarp/CMakeFiles/lokinet-util.dir/util/lokinet_init.c.o
[ 40%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/mem.cpp.o
[ 40%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/printer.cpp.o
[ 41%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/str.cpp.o
[ 41%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/thread/queue_manager.cpp.o
[ 43%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/time.cpp.o
[ 43%] Building CXX object llarp/CMakeFiles/lokinet-util.dir/util/thread/threading.cpp.o
[ 43%] Linking CXX static library liblokinet-util.a
[ 43%] Built target lokinet-util
[ 44%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/ev/ev.cpp.o
[ 44%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_address.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/ev/ev_libuv.cpp.o
[ 45%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_packet.cpp.o
[ 46%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/net.cpp.o
[ 46%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/ip_range.cpp.o
[ 46%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/net_int.cpp.o
[ 47%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/packet_router.cpp.o
[ 47%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/egres_packet_router.cpp.o
[ 47%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/net/sock_addr.cpp.o
[ 47%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/vpn/platform.cpp.o
[ 48%] Building CXX object llarp/CMakeFiles/lokinet-platform.dir/linux/netns.cpp.o
[ 48%] Linking CXX static library liblokinet-platform.a
[ 48%] Built target lokinet-platform
[ 48%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/ini.cpp.o
[ 48%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/config.cpp.o
[ 48%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/key_manager.cpp.o
[ 50%] Building CXX object llarp/CMakeFiles/liblokinet.dir/config/definition.cpp.o
[ 51%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/message.cpp.o
[ 51%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/name.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/rr.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/question.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/serialize.cpp.o
[ 52%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/server.cpp.o
[ 53%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/srv_data.cpp.o
[ 53%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dns/unbound_resolver.cpp.o
[ 53%] Building CXX object llarp/CMakeFiles/liblokinet.dir/consensus/table.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/consensus/reachability_testing.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/bootstrap.cpp.o
[ 54%] Building CXX object llarp/CMakeFiles/liblokinet.dir/context.cpp.o
[ 55%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/crypto_libsodium.cpp.o
[ 55%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/crypto.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/encrypted_frame.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/crypto/types.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/context.cpp.o
[ 56%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/dht.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/explorenetworkjob.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localtaglookup.cpp.o
[ 58%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localrouterlookup.cpp.o
[ 59%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/localserviceaddresslookup.cpp.o
[ 59%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findintro.cpp.o
[ 59%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/message.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findrouter.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotintro.cpp.o
[ 60%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotrouter.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/findname.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/pubintro.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/messages/gotname.cpp.o
[ 61%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/publishservicejob.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/recursiverouterlookup.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/taglookup.cpp.o
[ 62%] Building CXX object llarp/CMakeFiles/liblokinet.dir/dht/serviceaddresslookup.cpp.o
[ 63%] Building CXX object llarp/CMakeFiles/liblokinet.dir/endpoint_base.cpp.o
[ 63%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/context.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/exit_messages.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/endpoint.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/policy.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/handlers/tun.cpp.o
[ 65%] Building CXX object llarp/CMakeFiles/liblokinet.dir/exit/session.cpp.o
[ 66%] Building CXX object llarp/CMakeFiles/liblokinet.dir/handlers/exit.cpp.o
[ 66%] Building CXX object llarp/CMakeFiles/liblokinet.dir/hook/shell.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/iwp.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/linklayer.cpp.o
[ 67%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/message_buffer.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/iwp/session.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/session.cpp.o
[ 68%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/link_manager.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/link/server.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/dht_immediate.cpp.o
[ 69%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/link_intro.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/link_message_parser.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay.cpp.o
[ 70%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay_commit.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/messages/relay_status.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/address_info.cpp.o
[ 72%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/exit_info.cpp.o
[ 73%] Building CXX object llarp/CMakeFiles/liblokinet.dir/net/traffic_policy.cpp.o
[ 73%] Building CXX object llarp/CMakeFiles/liblokinet.dir/nodedb.cpp.o
[ 73%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/ihophandler.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/path_context.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/path.cpp.o
[ 74%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/pathbuilder.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/pathset.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/path/transit_hop.cpp.o
[ 75%] Building CXX object llarp/CMakeFiles/liblokinet.dir/peerstats/peer_db.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/peerstats/types.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/pow.cpp.o
[ 76%] Building CXX object llarp/CMakeFiles/liblokinet.dir/profiling.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/address.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/client.cpp.o
[ 77%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/connection.cpp.o
[ 79%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/null_crypto.cpp.o
[ 79%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/endpoint.cpp.o
[ 79%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/server.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/stream.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/quic/tunnel.cpp.o
[ 80%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_contact.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_id.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router_version.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/name.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/outbound_message_handler.cpp.o
[ 81%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/rc_lookup_handler.cpp.o
[ 82%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/outbound_session_maker.cpp.o
[ 82%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/rc_gossiper.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/router.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/systemd_resolved.cpp.o
[ 83%] Building CXX object llarp/CMakeFiles/liblokinet.dir/router/route_poker.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/dht_message.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/message_parser.cpp.o
[ 84%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_confirm_message.cpp.o
[ 86%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_latency_message.cpp.o
[ 86%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/path_transfer_message.cpp.o
[ 86%] Building CXX object llarp/CMakeFiles/liblokinet.dir/routing/transfer_traffic_message.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/rpc_server.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/lokid_rpc_client.cpp.o
[ 87%] Building CXX object llarp/CMakeFiles/liblokinet.dir/rpc/endpoint_rpc.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/address.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/async_key_exchange.cpp.o
[ 88%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/auth.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/convotag.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/context.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint_state.cpp.o
[ 89%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint.cpp.o
[ 90%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/endpoint_util.cpp.o
[ 90%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/hidden_service_address_lookup.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/identity.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/info.cpp.o
[ 91%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/intro_set.cpp.o
[ 93%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/intro.cpp.o
[ 93%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/lns_tracker.cpp.o
[ 93%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/lookup.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/outbound_context.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/protocol.cpp.o
[ 94%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/protocol_type.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/router_lookup_job.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/sendcontext.cpp.o
[ 95%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/session.cpp.o
[ 96%] Building CXX object llarp/CMakeFiles/liblokinet.dir/service/tag.cpp.o
[ 96%] Linking CXX static library liblokinet.a
[ 96%] Built target liblokinet
[ 96%] Building CXX object llarp/CMakeFiles/lokinet-shared.dir/lokinet_shared.cpp.o
[ 96%] Building CXX object daemon/CMakeFiles/lokinet.dir/lokinet.cpp.o
[ 96%] Building CXX object daemon/CMakeFiles/lokinet-bootstrap.dir/lokinet-bootstrap.cpp.o
[ 96%] Building CXX object daemon/CMakeFiles/lokinet-vpn.dir/lokinet-vpn.cpp.o
[ 96%] Linking CXX executable lokinet-bootstrap
[ 97%] Linking CXX executable lokinet
[ 98%] Linking CXX shared library liblokinet.so
[100%] Linking CXX executable lokinet-vpn
[100%] Built target lokinet-bootstrap
[100%] Built target lokinet-vpn
[100%] Built target lokinet-shared
[100%] Built target lokinet
~> build (issue-1911-fixup-oxenc-linkage) sudo make install
Consolidate compiler generated dependencies of target oxenmq
[  3%] Built target oxenmq
Consolidate compiler generated dependencies of target lokinet-cryptography
[ 16%] Built target lokinet-cryptography
[ 16%] Built target genversion
Consolidate compiler generated dependencies of target lokinet-util
[ 24%] Built target lokinet-util
Consolidate compiler generated dependencies of target lokinet-platform
[ 30%] Built target lokinet-platform
Consolidate compiler generated dependencies of target ngtcp2_static
[ 43%] Built target ngtcp2_static
Consolidate compiler generated dependencies of target liblokinet
[ 90%] Built target liblokinet
Consolidate compiler generated dependencies of target lokinet-shared
[ 91%] Built target lokinet-shared
Consolidate compiler generated dependencies of target lokinet-vpn
[ 93%] Built target lokinet-vpn
Consolidate compiler generated dependencies of target lokinet
[ 94%] Built target lokinet
Consolidate compiler generated dependencies of target cpr
[100%] Built target cpr
Consolidate compiler generated dependencies of target lokinet-bootstrap
[100%] Built target lokinet-bootstrap
Install the project...
-- Install configuration: "Release"
-- Installing: /usr/local/lib64/liboxenmq.so.0
-- Installing: /usr/local/lib64/liboxenmq.so
-- Installing: /usr/local/include/oxenmq/address.h
-- Installing: /usr/local/include/oxenmq/auth.h
-- Installing: /usr/local/include/oxenmq/base32z.h
-- Installing: /usr/local/include/oxenmq/base64.h
-- Installing: /usr/local/include/oxenmq/batch.h
-- Installing: /usr/local/include/oxenmq/bt_serialize.h
-- Installing: /usr/local/include/oxenmq/bt_value.h
-- Installing: /usr/local/include/oxenmq/connections.h
-- Installing: /usr/local/include/oxenmq/hex.h
-- Installing: /usr/local/include/oxenmq/oxenmq.h
-- Installing: /usr/local/include/oxenmq/message.h
-- Installing: /usr/local/include/oxenmq/variant.h
-- Installing: /usr/local/include/oxenmq/version.h
-- Installing: /usr/local/include/oxenmq/zmq.hpp
-- Installing: /usr/local/lib64/pkgconfig/liboxenmq.pc
-- Installing: /usr/local/include/lokimq/address.h
-- Installing: /usr/local/include/lokimq/auth.h
-- Installing: /usr/local/include/lokimq/base32z.h
-- Installing: /usr/local/include/lokimq/base64.h
-- Installing: /usr/local/include/lokimq/batch.h
-- Installing: /usr/local/include/lokimq/bt_serialize.h
-- Installing: /usr/local/include/lokimq/bt_value.h
-- Installing: /usr/local/include/lokimq/connections.h
-- Installing: /usr/local/include/lokimq/hex.h
-- Installing: /usr/local/include/lokimq/lokimq.h
-- Installing: /usr/local/include/lokimq/message.h
-- Installing: /usr/local/include/lokimq/variant.h
-- Installing: /usr/local/include/lokimq/version.h
-- Installing: /usr/local/lib64/pkgconfig/liblokimq.pc
-- Installing: /usr/local/lib64/liblokinet.so
-- Set runtime path of "/usr/local/lib64/liblokinet.so" to ""
-- Installing: /usr/local/bin/lokinet
-- Set runtime path of "/usr/local/bin/lokinet" to ""
-- Installing: /usr/local/bin/lokinet-vpn
-- Set runtime path of "/usr/local/bin/lokinet-vpn" to ""
-- Installing: /usr/local/bin/lokinet-bootstrap
-- Set runtime path of "/usr/local/bin/lokinet-bootstrap" to ""
</pre></code>
</details>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -199,6 +199,7 @@ if(OXENC_FOUND)
 else()
   message(STATUS "using oxen-encoding submodule")
   add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/external/oxen-encoding)
+  add_library(oxenc::oxenc ALIAS oxenc)
 endif()
 
 option(FORCE_OXENMQ_SUBMODULE "force using oxenmq submodule" OFF)

--- a/llarp/CMakeLists.txt
+++ b/llarp/CMakeLists.txt
@@ -34,7 +34,7 @@ target_link_libraries(lokinet-util PUBLIC
   nlohmann_json::nlohmann_json
   filesystem
   date::date
-  oxenmq::oxenmq
+  oxenc::oxenc
 )
 
 if(ANDROID)
@@ -59,7 +59,7 @@ add_library(lokinet-platform
 )
 
 target_link_libraries(lokinet-platform PUBLIC lokinet-cryptography lokinet-util Threads::Threads base_libs uvw)
-
+target_link_libraries(lokinet-platform PRIVATE oxenmq::oxenmq)
 
 if (ANDROID)
   target_sources(lokinet-platform PRIVATE android/ifaddrs.c)
@@ -242,7 +242,7 @@ if(WITH_HIVE)
   )
 endif()
 
-target_link_libraries(liblokinet PUBLIC cxxopts lokinet-platform lokinet-util lokinet-cryptography sqlite_orm ngtcp2_static)
+target_link_libraries(liblokinet PUBLIC cxxopts oxenc::oxenc lokinet-platform lokinet-util lokinet-cryptography sqlite_orm ngtcp2_static oxenmq::oxenmq)
 target_link_libraries(liblokinet PRIVATE libunbound)
 pkg_check_modules(CRYPT libcrypt IMPORTED_TARGET)
 if(CRYPT_FOUND AND NOT CMAKE_CROSSCOMPILING)

